db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -4125,7 +4125,14 @@ Status DBImpl::WaitForCompact(
          bg_flush_scheduled_ || unscheduled_compactions_ ||
          unscheduled_flushes_ || error_handler_.IsRecoveryInProgress()) &&
         (error_handler_.GetBGError().ok())) {
-      bg_cv_.Wait();
+      if (wait_for_compact_options.timeout_micros) {
+        if (bg_cv_.TimedWait(immutable_db_options_.clock->NowMicros() +
```

Comment:
IIUC, this logic will recalculate the `wait_util_time` everytime this block is entered. For example, when one compaction job finishes and signals,  this `TimedWait` can get notified and wake up, assuming it's not timed out yet, it will enter the next round of `for(;;)` looping. At that time, it will get the `now` time again to recalculate the `wait_util` time.  Is that correct? In that case, you probably would need to get the `wait_util` time before entering the for loop and use it instead.

Reply:
Ahhh great point. Yeah, this will recalculate the timeout in every loop indeed. Let me fix it.